plugins/flatpak/gs-flatpak.c

```diff
@@ -562,20 +576,51 @@ gs_flatpak_mark_apps_from_usb_remote (GsFlatpak *self,
 	for (guint i = 0; i < refs->len; ++i) {
 		FlatpakRef *ref = FLATPAK_REF (g_ptr_array_index (refs, i));
 		g_autoptr(GsApp) app = gs_flatpak_create_app (self, NULL /* origin */, ref);
+#else
+	/* In Flatpak 1.7.1 and later
+	 * flatpak_installation_list_remote_refs_sync() no longer works for
+	 * sideload repos due to https://github.com/flatpak/flatpak/pull/3476
+	 */
+	g_autoptr(GHashTable) refs = NULL;
+	g_autoptr(GFile) repo_file = g_file_new_for_uri (remote_url);
+	g_autoptr(OstreeRepo) repo = ostree_repo_new (repo_file);
+	g_autoptr(GPtrArray) remotes = NULL;
+	GHashTableIter iter;
+	gpointer key;
+
+	if (!ostree_repo_open (repo, cancellable, error))
+		return FALSE;
+
+	if (!ostree_repo_list_collection_refs (repo, NULL, &refs, 0, cancellable, error))
+		return FALSE;
```

Comment:
Strictly speaking no, since the error will only be printed in a debug message not passed along to other parts of g-s, but we can do it anyway in case this code gets rearranged.